src/pages/apps/[systemId]/[appId]/edit.js

```diff
@@ -42,39 +42,33 @@ export default function AppEdit() {
 
     const { isFetching: appInfoLoading } = useQuery({
         queryKey: [APP_BY_ID_QUERY_KEY, { systemId, appId }],
-        queryFn: getAppById,
-        config: {
-            enabled: userProfile?.id && systemId && appId,
-            onSuccess: (result) => {
-                setAppListingInfo(result?.apps[0]);
-            },
-            onError: setLoadingError,
+        queryFn: () => getAppById({ systemId, appId }),
+        enabled: userProfile?.id && systemId && appId,
```

Comment:
This needs to be a boolean:
```suggestion
        enabled: !!(userProfile?.id && systemId && appId),
```